upf/upf_cli.c

```diff
@@ -253,15 +253,12 @@ upf_nat_pool_add_del_command_fn (vlib_main_t * vm,
   u8 *nwi_name;
   u8 *nwi_s = 0;
   ip4_address_t start, end;
-  u16 min_port;
-  u16 max_port;
-  u16 port_block_size;
+  u32 min_port = UPF_NAT_MIN_PORT;
+  u32 max_port = UPF_NAT_MAX_PORT;
```

Comment:
`vnet_upf_nat_pool_add_del` takes u16 for min, max and block size. Passing the u32 values into it will truncate the high bits.
The CLI is debug tool, so I don't really mind this. But it might not be what you had in mind. Checking the values for a 16bit overflow might be sensible.

Reply:
Added extra check